Smart Contract

```diff
@@ -70,37 +69,36 @@ contract Auction {
     function Auction ();
     
     function createNewAuction (uint _auctionTime, uint _minimumBid ) returns (uint) {
-        uint AuctionID = NumberOfAuctions.length;
-        NumberOfAuctions.push(1);
+        uint AuctionID = numberOfAuctions;
+        numberOfAuctions = numberOfAuctions + 1;
         endAuctionTime[AuctionID] = now + _auctionTime;
         minimumBid[AuctionID] = _minimumBid;
         auctionEnded[AuctionID] = false; 
         seller[AuctionID] = msg.sender;
         return AuctionID;
     }
     
-    function bid (uint _amount, uint AuctionID) payable isOver (AuctionID) {
-        require (_amount > highestBid[AuctionID] && _amount > minimumBid[AuctionID]);
-        HighestBidIncreased(msg.sender, _amount);
-        refundableBids[AuctionID][highestBidder[AuctionID]] += highestBid[AuctionID];
-        highestBid[AuctionID] = _amount;
+    function bid (uint AuctionID) payable isOver (AuctionID) {
+        require (msg.value > highestBid[AuctionID] && msg.value > 2 * minimumBid[AuctionID]);
+        HighestBidIncreased(msg.sender, msg.value);
+        refundableBids[highestBidder[AuctionID]] += highestBid[AuctionID];
+        highestBid[AuctionID] = msg.value;
         highestBidder[AuctionID] = msg.sender;
     }
     
     function withdrawRefundableBid (uint AuctionID) {
         require (seller[AuctionID] != msg.sender);
-        uint refund = refundableBids[AuctionID][msg.sender];
+        uint refund = refundableBids[msg.sender];
         require (refund > 0);
-        refundableBids[AuctionID][msg.sender] = 0;
+        refundableBids[msg.sender] = 0;
         msg.sender.transfer(refund);
         
     }
     
-    function endAuction (uint AuctionID) {
+    function endAuction (uint AuctionID) isOver(AuctionID) {
         require (auctionEnded[AuctionID] == false);
-        require (now > endAuctionTime[AuctionID]);
         auctionEnded[AuctionID] = true; 
-        if (highestBid >= minimumBid) {
+        if (highestBid[AuctionID] >= 2 * minimumBid[AuctionID]) {
```

Comment:
This is something that you are enforcing elsewhere, and so does not need to be checked here.

Usually, it's not bad practice to assert things multiple times to make sure you can't get into a really bad state, but I think this is simple enough it's not worth it.